React/Modules/RCTTiming.m

```diff
@@ -181,22 +192,31 @@ - (void)didUpdateFrame:(__unused RCTFrameUpdate *)update
   // Call timers that need to be called
   if (timersToCall.count > 0) {
     [_bridge enqueueJSCall:@"JSTimersExecution.callTimers" args:@[timersToCall]];
-
-    // If we call at least one timer this frame, don't switch to a paused state yet, so if
-    // in response to this timer another timer is scheduled, we don't pause and unpause
-    // the displaylink frivolously.
-    return;
   }
 
-  // No need to call the pauseCallback as RCTDisplayLink will ask us about our paused
-  // status immediately after completing this call
-  if (_timers.count == 0) {
-    _paused = YES;
+  if (_sendIdleEvents) {
+    NSTimeInterval frameElapsed = (CACurrentMediaTime() - update.timestamp) * 1000;
+    NSTimeInterval currentTimestamp = [[NSDate date] timeIntervalSince1970] * 1000;
+    NSNumber *absoluteFrameStartMillis = [NSNumber numberWithDouble:currentTimestamp - frameElapsed];
```

Comment:
Yeah. The way I think of this is: when working in Obj-C/Cocoa we want to follow Cocoa conventions most of the time (in this case, use `NSTimeInterval` and define the values in seconds, which is what people expect NSTimeInterval to represent). And then, as late as possible, convert from seconds to milliseconds when we are sending these values from Obj-C to JS.